### PR TITLE
Opt-in to rustc_private for `rust-analyzer`

### PR DESCRIPTION
**Pull request by DJMcNab**
rust-analyzer/rust-analyzer#7891

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -23,3 +23,7 @@ serde_json = "1.0.64"
 
 [dev-dependencies]
 quickcheck = "1.0"
+
+[package.metadata.rust-analyzer]
+# This crate uses #[feature(rustc_private)]
```
Comment (JohnTitor) [inline Cargo.toml:28]:
```suggestion
# This crate uses `#![feature(rustc_private)]`
# c.f. https://github.com/rust-analyzer/rust-analyzer/pull/7891
```

**Comment by JohnTitor** [comment]
Merged via 32f84042cd62ae7176a4e010ec608d1a502a3881.

**Comment by Xanewok** [comment]
Huh, that's weird. Maybe worth reopening to see if this time bors feels okay?

**Comment by DJMcNab** [comment]
It appears to be merged anyway

**Comment by Xanewok** [comment]
Oh, you're right. Thanks again!

**Pull request closed without merging**